### DISCUSSION TITLE
TypeError: ufunc 'bs_delta' not supported for the input types

### DISCUSSION THREAD

**Discussion by zednotz** [domokane/FinancePy#201] [Q&A] [answered]
Hi, I'm trying to create a greeks pricer. 

I'm facing an Error and I'm not figuring it out.

### here's a part of my code : 

import numpy as np
from financepy.models.black_scholes_analytic import bs_delta, bs_gamma, bs_vega, bs_theta, bs_rho, bs_vanna, bs_implied_volatility
from financepy.utils.global_types import OptionTypes, FinLongShort, FinExerciseTypes

    def calculate_greeks():
      s = np.float64(entry_S.get())  
      k = np.float64(entry_K.get())  
      t = np.float64(entry_T.get())  
      r = np.float64(entry_r.get())  
      v = np.float64(entry_sigma.get())  
      q = np.float64(entry_q.get())
      q_str = entry_q.get()
      if q_str == "":
          q = 0.0  # Si le champ est vide, traitez-le comme zéro
      else:
          q = np.float64(q_str)
      option_type_str = option_type_var.get()
      option_direction_str = option_direction_var.get()
      option_style_str = option_style_var.get()
    
    # Accédez à l'énumération OptionTypes, FinLongShort et FinExerciseTypes
    option_type_value = OptionTypes[option_type_str]
    option_direction = FinLongShort[option_direction_str]
    exercise_type = FinExerciseTypes.AMERICAN if option_style_str == "AMERICAN" else FinExerciseTypes.EUROPEAN

    delta = bs_delta(s, t, k, r, 0, v, option_type_value)
    gamma = bs_gamma(s, t, k, r, 0, v, option_type_value)
    vega = bs_vega(s, t, k, r, 0, v, option_type_value)
    theta = bs_theta(s, t, k, r, 0, v, option_type_value)
    rho = bs_rho(s, t, k, r, 0, v, option_type_value)
    vanna = bs_vanna(s, t, k, r, 0, v, option_type_value)
    implied_volatility = bs_implied_volatility(s, t, k, r, 0, option_type_value)

    label_result.config(text=f"Delta: {delta:.4f}\nGamma: {gamma:.4f}\nVega: {vega:.4f}\nTheta: {theta:.4f}\nRho: {rho:.4f}\nVanna: {vanna:.4f}\nVolatilité implicite: {implied_volatility:.4f}")
    
### Here's the error : 

Exception in Tkinter callback
Traceback (most recent call last):
  File "/Users/BUSINESS/anaconda3/lib/python3.10/tkinter/__init__.py", line 1921, in __call__
    return self.func(*args)
  File "/Users/BUSINESS/Desktop/PRICER GREC.py", line 34, in calculate_greeks
    delta = bs_delta(s, t, k, r, 0, v, option_type_value)
TypeError: ufunc 'bs_delta' not supported for the input types, and the inputs could not be safely coerced to any supported types according to the casting rule ''safe''

I've tried a lot, convert with np.float64 etc.. chatgpt doesn't seem to wanna help me xD

**Comment by zednotz** [comment]
I figured it out ! Thanks

**Reply by domokane** [reply]
You need to convert option type to integer value ?

**Comment by domokane** [comment]
What was your fix ?

**Reply by zednotz** [reply, marked as answer]
instead of : 

 option_type_value = OptionTypes[option_type_str]
 option_direction = FinLongShort[option_direction_str]

I added '.value' at the end : 

 option_type_value = OptionTypes[option_type_str.upper()].value
 option_direction = FinLongShort[option_direction_str.upper()].value

**Comment by domokane** [comment]
Yes. As I thought. Tell me how it goes. Happy to get feedback and to know who is using financepy and why.
